### PR TITLE
686 marriages

### PR DESCRIPTION
**Pull request by jholton**
Changes to veteran's marriages
The second commit on this PR is a refactor, so to see the substantive changes check out the first commit:  https://github.com/department-of-veterans-affairs/vets-json-schema/pull/270/commits/3b22e46e4749016a34f0c01263637a3b438744eb

Basically, I separated out `marriages` into `currentMarriage` and `previousMarriages` since the field requirements are slightly different (`ssn` necessary for `currentMarriage`; marriage termination details necessary for `previousMarriages`)

### CODE REVIEW

**Comment by jholton** [comment]
The second commit is a refactor where I moved as much as made sense from `const` variables into the schema's `definitions` property.

File: src/schemas/21-686C/schema.js [hunk 1]
```diff
@@ -285,129 +285,228 @@ const commonAddressFields = {
...
+        type: 'array',
+        items: {
+          type: 'object',
+          required: [...commonMarriageDef.required, 'reasonForSeparation', 'dateOfSeparation', 'locationOfSeparation'],
+          required: ['reasonForSeparation'],
+          properties: {
+            ...commonMarriageDef.properties,
+            reasonForSeparation: {
+              type: 'string',
+              'enum': [
```
Comment (jholton) [inline src/schemas/21-686C/schema.js]: Enum values taken from `EVSS DAM Form 686c_674 Updated version 17_14`

File: src/schemas/21-686C/schema.js [hunk 2]
```diff
@@ -464,72 +572,25 @@ let schema = {
...
+          },
+          personWhoLivesWithChild: schemaHelpers.getDefinition('fullName')
         }
       }
     }
   },
-  required: ['privacyAgreementAccepted']
+  required: [
+    'privacyAgreementAccepted',
+
```
Comment (jholton) [inline src/schemas/21-686C/schema.js]: Oops, meant to fill this out.

File: src/schemas/21-686C/schema.js [hunk 3]
```diff
@@ -464,72 +572,27 @@ let schema = {
...
       }
     }
   },
-  required: ['privacyAgreementAccepted']
+  required: [
+    'privacyAgreementAccepted',
+    'veteranFullName',
+    'veteranAddress',
+    'maritalStatus'
+  ]
```
Comment (jholton) [inline src/schemas/21-686C/schema.js]: How should we handle requirement of `ssn` and/or `file number`?

Comment (lihanli) [reply src/schemas/21-686C/schema.js]: https://github.com/department-of-veterans-affairs/vets-json-schema/blob/master/src/schemas/21P-527EZ/schema.js#L18

File: src/schemas/21-686C/schema.js [hunk 4]
```diff
@@ -285,129 +285,228 @@ const commonAddressFields = {
...
+  { $ref: '#/definitions/internationalAddressText' }
+]
 
 let schema = {
   $schema: 'http://json-schema.org/draft-04/schema#',
   title: 'DECLARATION OF STATUS OF DEPENDENTS',
   type: 'object',
   additionalProperties: false,
-  definitions,
+  definitions: _.merge(_.pick(definitions, 'fullName'),
```
Comment (lihanli) [inline src/schemas/21-686C/schema.js]: `_.pick(definitions, ['fullName'])` is already being called above, how about calling `_.merge(definitions, ...)` here?

File: src/schemas/21-686C/schema.js (hunk 3 above)
Comment (jholton) [reply src/schemas/21-686C/schema.js]: Thanks

**Comment by lihanli** [comment]
can you update the version?

**Comment by jholton** [comment]
Merging this one, so I can get back to working on prefill.  There's a good chance there will be another one to address the `remarks` field.

**Pull request merged**